scripts/process/grid_from_mask.R

```diff
@@ -0,0 +1,21 @@
+#' take an sf object polygon and make a grid out of it
+#' 
+#' 
+process.grid_from_mask <- function(viz){
+  deps <- readDepends(viz)
+  checkRequired(viz, c("cell_size"))
+  
+  clip_poly <- deps[['clip_poly']]
+  crs <- sf::st_crs(clip_poly)
+  cell_size <- viz[["cell_size"]]
+  
+  # this is a simple way to dissolve multiple polygons into a single mask
+  clip_mask <- clip_poly %>% 
+    mutate(group=1) %>% group_by(group) %>% summarise()
+  cell_grid <- sf::st_make_grid(clip_mask, cellsize = cell_size, crs = crs)
+  cell_grid_clipped <- sf::st_intersection(cell_grid, clip_mask)
```

Comment:
In the past we've also wanted the cell centroids... Not sure if there is resistance to creating a list containing two or three "versions" of these cells as output, but the unclipped version, the clipped version, and the cell centers for all cells that are partially in the clipped version might be handy?